platform/android/os_android.h

```diff
@@ -200,24 +158,72 @@ class OS_Android : public OS_Unix {
 	void process_gravity(const Vector3 &p_gravity);
 	void process_magnetometer(const Vector3 &p_magnetometer);
 	void process_gyroscope(const Vector3 &p_gyroscope);
-	void process_touch(int p_event, int p_pointer, const Vector<TouchPos> &p_points);
-	void process_hover(int p_type, Point2 p_pos);
-	void process_mouse_event(int event_action, int event_android_buttons_mask, Point2 event_pos, float event_vertical_factor = 0, float event_horizontal_factor = 0);
-	void process_double_tap(int event_android_button_mask, Point2 p_pos);
-	void process_scroll(Point2 p_pos);
-	void process_joy_event(JoypadEvent p_event);
-	void process_key_event(int p_keycode, int p_scancode, int p_unicode_char, bool p_pressed);
-	void process_event(Ref<InputEvent> p_event);
 	void init_video_mode(int p_video_width, int p_video_height);
 
 	virtual bool is_joy_known(int p_device);
 	virtual String get_joy_guid(int p_device) const;
```

Comment:
Should these methods also be moved to the `AndroidInputHandler` class?

Reply:
My criteria for moving functions to `AndroidInputHandler` has been the following:
- They were not overrides of `OS`'s functions, since those are only guaranteed to be safe on the Android render thread (Godot main thread) and therefore don't need an special treatment.
- They will be called from the Android UI thread.
- They only call thread-safe functions.
All the points have turned out to be met at the same time for all the functions of interest.